src/components/perception.rs

```diff
@@ -0,0 +1,21 @@
+use amethyst::{
+    assets::{PrefabData, PrefabError},
+    ecs::{Component, DenseVecStorage, Entity, WriteStorage},
+};
+use amethyst_inspector::Inspect;
+use serde::{Deserialize, Serialize};
+
+#[derive(Default, Clone, Debug, Inspect, Serialize, Deserialize, PrefabData)]
+#[prefab(Component)]
+#[serde(default)]
+pub struct Perception {
+    pub range: f32,
+
+    #[serde(skip)]
+    #[inspect(skip)]
+    pub entities: Vec<Entity>,
```

Comment:
I'd prefer to have this split into a separate component. Maybe `Range` and `PerceivedEntities`. I think we might change `PerceivedEntities` in the feature. Maybe to something like `Perceived<T>` where we collect perceived entities based on tags

Reply:
I split it into the `Perception` component with the `range`value and a `DetectedEntities` component.